src/controllers/main/main.ts

```diff
@@ -1182,7 +1182,20 @@ export class MainController extends EventEmitter {
         ) as AccountOpAction | undefined
 
         const hasAuthorized = !!currentAccountOpAction?.accountOp?.meta?.entryPointAuthorization
-        if (shouldAskForEntryPointAuthorization(network, account, accountState, hasAuthorized)) {
+
+        this.activity.setFilters({
+          account: account.addr,
+          network: network.id
+        })
+        const entryPointAuthorizationMessageFromHistory = this.activity.signedMessages?.items.find(
+          (message) =>
+            message.fromActionId === ENTRY_POINT_AUTHORIZATION_REQUEST_ID &&
+            message.networkId === network.id
+        )
+        if (
+          shouldAskForEntryPointAuthorization(network, account, accountState, hasAuthorized) &&
+          !entryPointAuthorizationMessageFromHistory
```

Comment:
We can mark the `hasAuthorized` as true if `entryPointAuthorizationMessageFromHistory` is present and therefore skip the extra `!entryPointAuthorizationMessageFromHistory` check